Sources/Services/ContainerImagesService/Server/ImageService.swift

```diff
@@ -166,7 +166,13 @@ extension ImagesService {
             return try await body(authentication)
         }
         let keychain = KeychainHelper(id: Self.keychainID)
-        authentication = try? keychain.lookup(domain: host)
+        do {
+            authentication = try keychain.lookup(domain: host)
+        } catch let err as KeychainHelper.Error {
+            guard case .keyNotFound = err else {
+                throw ContainerizationError(.internalError, message: "\(String(describing: err)). Error querying keychain for \(host)")
```

Comment:
`ContainerizationError` has a `cause` field that you can include the original error on:
```swift
throw ContainerizationError(.internalError, message: "Error querying keychain for \(host)", cause: err)
```

Reply:
We also don't handle any of the other ways `.get()` and `.exists()` can fail internally in `.lookup()`, this only catches `.keyNotFound`.

Reply:
We intentionally only want to catch the keyNotFound error, since we may still be able to pull/push the image in that scenario.